# Bioinformatics
(CMPSC 300) Lab 4

This repository contains information about Bioinformatics Activity 04 for 5th and 7th Oct 2022. This activity invites students to improve Biopython code and then to prepare evidence of __Genetic Drift__.

## Dates

Handed out: 05 Oct 2022

Due: 07 Oct 2022

## Instructions

### Biopython code

+ Locate the Biopython sequence analysis tool (`bpAlign_iii.py`) in `src/` to run.  

+ Read the instructions at the top of the source code to create and launch a virtual environment.
__Note, to avoid having to send a lot of data (about 98M in total) to GitHub, please consider not creating your virtual environment file system in your GitHub Repository__.
Remember, you can always build the virtual environment in a local working directory and then activate it there for each time you need it.

+ Install Biopython in this virtual environment.

+ Create some sample data (sequences and a fasta file) to be used to test this code.

+ Improve the code: From your previous lab, or from another origin, can you implement a new test into this code to provide more study?
Or, could you improve the code by adding functionality to process more than two sequences at a time by file input or interactively?

### Genetic Drift Study

+ Select any ten related nucleotide sequences from [NCBI](https://www.ncbi.nlm.nih.gov/).

+ Prepare this data in files or as sequences to be used with your improved tool to provide  proof for __Genetic Drift__ across species. Your study can use any related sequences.

+ Address `writing/reflection.md`
  
### Present your work

+ Come to class with about two slides to explain your improvements to the tool and to discuss your experiments.

## Assessment

This grade is a check mark.
